server_addon/houdini/client/ayon_houdini/plugins/load/load_hda.py

```diff
@@ -18,38 +39,44 @@ class HdaLoader(plugin.HoudiniLoader):
     color = "orange"
 
     def load(self, context, name=None, namespace=None, data=None):
-        import hou
 
         # Format file name, Houdini only wants forward slashes
         file_path = self.filepath_from_context(context)
         file_path = os.path.normpath(file_path)
         file_path = file_path.replace("\\", "/")
 
-        # Get the root node
-        obj = hou.node("/obj")
-
         # Create a unique name
         counter = 1
         namespace = namespace or context["folder"]["name"]
         formatted = "{}_{}".format(namespace, name) if namespace else name
         node_name = "{0}_{1:03d}".format(formatted, counter)
 
+        hda_defs = hou.hda.definitionsInFile(file_path)
+        if not hda_defs:
+            raise RuntimeError ("No HDA definitions found!")
+
+        hda_def = hda_defs[0]
+        parent_node = self._create_dedicated_parent_node(hda_def)
+
         hou.hda.installFile(file_path)
-        hda_node = obj.createNode(name, node_name)
+        hda_node = parent_node.createNode(name, node_name)
+        hda_node.moveToGoodPosition()
 
-        self[:] = [hda_node]
+        # Imprint it manually
+        data = {
+            "schema": "openpype:container-2.0",
+            "id": AVALON_CONTAINER_ID,
+            "name": node_name,
+            "namespace": namespace,
+            "loader": str(self.__class__.__name__),
+            "representation": context["representation"]["id"],
+        }
```

Comment:
Just wondering why do we imprint the data here instead of using `containerise` function? 
if it is just adding attributes with the data into the nodes, guess we can improve the `containerise` function instead of implementing the repetitive function?

Reply:
yeah, I had to do it because each HDA type needs to be loaded in a dedicated place. 

> Which makes me wonder a bit whether it's still worth it to put it inside that container group instead of trying to put it in an active matching panel or alike.

It's fair to support that. but I think it's something that's needs to be supported for other products too.
Maybe, I can add it here. but I'm afraid it becomes too much for this PR.

Reply:
Agree - likely best for another PR.

Reply:
I created an issue for it. and let's move the discussion there.
https://github.com/ynput/ayon-core/issues/650